### DISCUSSION TITLE
Is it (already) possible to sponsor bug fixes?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3826] [Questions & Answers]
Is it (already) possible to sponsor bug fixes?

**Comment by Flyingmana** [comment]
currently only if you finde one dev directly to solve it.

In the past there was https://bountysource.com/ but it seems to be offline right now.

We could make it indirectly over OpenCollective, but it will depend on the kind of bug, how much effort it is and so on, so currently I would still suggest to contact one of our listed freelancers directly.

**Reply by f1-outsourcing** [reply]
Yes I was already looking on [opencollective.com/openmage](https://opencollective.com/openmage). It would be nice if you could post an issue here, and then have it estimated / offered as paid service via [opencollective.com/openmage](https://opencollective.com/openmage)

**Comment by fballiano** [comment]
feel free to send me a message if you want.

**Reply by f1-outsourcing** [reply]
Send message where?

**Reply by fballiano** [reply]
if you click on my profile there are all my contacts

**Reply by f1-outsourcing** [reply]
ok

**Reply by f1-outsourcing** [reply]
Maybe check spam folder in gmail. This !#$@#% google/gmail is putting smaller email providers default in spam.

**Comment by f1-outsourcing** [comment]
The whole idea about having 'bugs' / 'enhancements' done with a financial compensation, is of course to create an environment where developers are doing less for free. The goal is to get more and faster bugs fixed without people having to do it for free. 
I can find a magento freelancer at standard rates over estimating the work any where.

Suggestion would be to have issues estimated by someone, and maintain a fixed hourly rate that is ok for most contributors. So if someone is willing to pay for the work any of the contributors can chose to do it. (Maybe spread this paid work among contributors, taking into account how much time they contribute)

**Reply by Flyingmana** [reply]
the downside of this would be, that the Maintainers basically acting as an agency then, needing to handle the contact with various Freelancers, trying to distribute the work fair, making sure the communication works properly, mediating in case of disputes.

In my personal view a direct relation between a company wanting to sponsor specific developments, and a freelancer would have a higher benefit as:
* communication works better in the long run
* communication works faster, as fewer people are involved
* provides additional opportunities for the freelancers to build Business relationships exceeding the Work on OpenMage itself (freelancers live from working with people who use OpenMage, not from working on OpenMage itself)

**Comment by f1-outsourcing** [comment]
> the downside of this would be, that the Maintainers basically acting as an agency then, needing to handle the contact with various Freelancers, trying to distribute the work fair, making sure the communication works properly, mediating in case of disputes.

What is there to mediate? If the issue is fixed and merged, what can the complaint be?

> 
> In my personal view a direct relation between a company wanting to sponsor specific developments, and a freelancer would have a higher benefit as:

But that is still the case, although not even necessary, because everything is described in the issue. 

>     * communication works better in the long run

obviously, but this is done via issue comments, not? Just like now.

>     * communication works faster, as fewer people are involved

obviously, but again this is done via issue comments. Just like now.

>     * provides additional opportunities for the freelancers to build Business relationships exceeding the Work on OpenMage itself (freelancers live from working with people who use OpenMage, not from working on OpenMage itself)

Yes, but that is not relevant here.

Currently it is not possible / difficult / cumbersome to sponsor issues. I think if you make that easier, it would be more utilized. Better for  OpenMage, better for OpenMage contributors and better for sponsors.

**Comment by jparker1986** [comment]
In my experience of bug fixes, it can be almost impossible to estimate. 

Sometimes it's a case of working with someone, paying their hourly rate and hoping it's not going to cost you the earth. 

The problem when working with multiple people is that it's not cost effective or productive and can lead to one developer fixing one problem and causing another, which the next person has to work out what the previous person did and so on.

**Comment by sreichel** [comment]
> freelancers live from working with people who use OpenMage, not from working on OpenMage itself

Good catch.

This should change a bit.

IMHO ...

If you want a specific bug (maybe 3rd-party) get fixed, report it, wait till its fixed an donate to oppencollective or personally.
If your are an agency/freelancer, you should really think about to fund this project. (!!!)
If your are an enduser, running OM on your own, contribute back with bug reports and possibly fixes. (Donations welcome o/c)

**Reply by f1-outsourcing** [reply]
> > freelancers live from working with people who use OpenMage, not from working on OpenMage itself
> 
> Good catch.

What is good about this? It is irrelevant. 


> This should change a bit.
> 
> IMHO ...
> 
> If you want a specific bug (maybe 3rd-party) get fixed, report it, wait till its fixed an donate to oppencollective or personally. If your are an agency/freelancer, you should really think about to fund this project. (!!!) If your are an enduser, running OM on your own, contribute back with bug reports and possibly fixes. (Donations welcome o/c)

Yes that is how it currently is ...
